### PR TITLE
test(expense): add hasReceipt query validator edge cases

### PR DESCRIPTION
**Pull request by ootakazuhiko**
## 概要
- `expenseListQuerySchema` の `hasReceipt` に対して、`1/0` 形式の受け入れを明示テスト化
- `TRUE` / 空白付き（`%20true%20`）を拒否するバリデーション境界を明示テスト化
- `hasReceipt` クエリ仕様の回帰検知を unit レイヤで補強

## 変更ファイル
- `packages/backend/test/expenseSchema.test.js`

## テスト
- `npm run test --prefix packages/backend`

### CODE REVIEW

File: packages/backend/test/expenseSchema.test.js [hunk 1]
```diff
@@ -212,6 +212,44 @@ test('expenseListQuerySchema: accepts settlement/receipt/date filters', async ()
...
+    url: '/validate/expense-list?hasReceipt=1',
+  });
+  assert.equal(oneRes.statusCode, 200);
+
+  const zeroRes = await app.inject({
+    method: 'GET',
+    url: '/validate/expense-list?hasReceipt=0',
+  });
+  assert.equal(zeroRes.statusCode, 200);
+  await app.close();
```
[the remark this answers is not in the document]
Comment (ootakazuhiko) [reply packages/backend/test/expenseSchema.test.js]: 対応しました。ご指摘どおり、失敗時に `app.close()` が実行されない可能性があるため、該当テストを `try/finally` で囲み常にクローズする形に修正しました（`f651e10`）。
